system/jlib/jsecrets.cpp

```diff
@@ -1441,21 +1501,18 @@ jlib_decl bool queryMtls()
                     if (checkFileExists(cert) && checkFileExists(privKey))
                     {
                         CriticalBlock block(mtlsInfoCacheCS);
-                        if (mtlsInfoCache)
-                        {
-                            IPropertyTree *info = mtlsInfoCache->queryPropTree("local");
-                            if (!info)
-                                info = mtlsInfoCache->setPropTree("local");
-                            if (info)
-                            {   // always update
-                                info->setProp("certificate", cert);
-                                info->setProp("privatekey", privKey);
-                                if ( (!isEmptyString(pubKey)) && (checkFileExists(pubKey)) )
-                                    info->setProp("publickey", pubKey);
-                                if (!isEmptyString(passPhrase))
-                                    info->setProp("passphrase", passPhrase); // encrypted
-                            }
-                        }
+                        assertex(mtlsInfoCache.find("local") == mtlsInfoCache.cend());
+
+                        Owned<IPropertyTree> info = createPTree("local");
+                        info->setProp("certificate", cert);
+                        info->setProp("privatekey", privKey);
+                        if ( (!isEmptyString(pubKey)) && (checkFileExists(pubKey)) )
+                            info->setProp("publickey", pubKey);
+                        if (!isEmptyString(passPhrase))
+                            info->setProp("passphrase", passPhrase); // encrypted
+
+                        Owned<ISyncedPropertyTree> entry = createSyncedPropertyTree(info);
```

Comment:
Not really new, although changing, but there may be a timing issue here because the meaning of "local" is being overridden.

In practical terms, looking up "local" means, get tls configuration based on the secret at "/opt/HPCCsystems/secrets/certificates/local".

The various ways of configuring bare metal need consideration.
One way of configuring bare metal could be to put TLS info into the secret locations we use in kubernetes. 
The other way of configuring bare-metal would be to fill in the custom TLS configuration values.

Here, a tree based on the custom bare-metal configuration values are turned into a tree that over rides the contents at the secret location.
If we ever (accidently?) use both the cached tree would be whichever came first.
So I guess we need to make sure if we are going to override "local" we always override "local".

Reply:
The alternative would be to cache under a different name, a name implying "bare-metal-mtls".  But we'd have to track down what was currently depending on it being called "local".